### PR TITLE
Wrap around

### PR DESCRIPTION
**Pull request by gdxn96**

### CODE REVIEW

File: Game/ConsoleApplication2/TerrainSegment.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{
+}
+
+void TerrainSegment::Update(float dt)
+{
+}
+
+void TerrainSegment::Draw(sf::RenderWindow & window)
+{
+	/*sf::RectangleShape r(sf::Vector2f(m_shape.getGlobalBounds().width, m_shape.getGlobalBounds().height));
```
Comment (ShaneCallanan) [inline Game/ConsoleApplication2/TerrainSegment.cpp]: this comment still needed?

File: Game/ConsoleApplication2/Terrain.cpp [hunk 2]
```diff
@@ -38,52 +38,54 @@ std::vector<int> calculateYValues(float minY, float maxY, float displace, float
...
-		r.draw(terrainShape);
-	}
-}
+	return points;
+}
+
+
+std::vector<TerrainSegment*> Terrain::GenerateTerrain(int minY, int maxY, Vector2D levelSize)
+{
+	sf::ConvexShape s;
```
Comment (ShaneCallanan) [inline Game/ConsoleApplication2/Terrain.cpp]: better variable names, or leave it as is for the moment?

File: Game/ConsoleApplication2/Terrain.cpp [hunk 3]
```diff
@@ -38,52 +38,54 @@ std::vector<int> calculateYValues(float minY, float maxY, float displace, float
...
+	float xOffset = static_cast<float>(levelSize.w) / yPoints.size();
+	sf::Vertex prev;
+	sf::Vertex current;
+	std::vector<TerrainSegment*> terrainSegments;
+	std::vector<sf::Vertex> vertices;
+
+	for (int i = 0; i < yPoints.size(); i++)
+	{
+		
+		current = sf::Vertex(sf::Vector2f(i * xOffset, yPoints[i]));
```
Comment (ShaneCallanan) [inline Game/ConsoleApplication2/Terrain.cpp]: same as above... could use 'currentVertex' - will be easier to scan later on

**Review by ShaneCallanan — DISMISSED** [review]
possible changes

**Comment by gdxn96** [comment]
Suggested changes done

**Review by ShaneCallanan — APPROVED** [review]

**Pull request merged**